### PR TITLE
Charles/update filters

### PR DESCRIPTION
**Pull request by Xhether**
## Overview

Finished Swap Filter Replacement 

### Change 1

- Added BRB's filter
- Added MealSwipes filter

### Change 2

- Removed payment methods filter
- Preserved respective order described in original task ( north, west, central, swipes, brbs, favorites, under 10 min )

## Test Coverage

- Compared results with filtering by brbs and mealswipes respectively from the main branch and my branch, which yielded the same expected results

## Next Steps (optional)

N/A

## Related PRs or Issues (optional)

N/a

## Screenshots (optional)

<details>
  <summary>TODO</summary>
  <table>
    <tr>
      <td>Before</td>
      <td>After</td>
    </tr>
  <tr>
    <td><video src="enterurlhere" type="video/mp4"></td>
    <td><video src="enterurlhere" type="video/mp4"></td>
  </tr>
 </table>
</details>

<details>
  <summary>TODO</summary>
  <table>
    <tr>
      <td>Before
</td>
      <td>After</td>
    </tr>
  <tr>
    <td></td>
    <td></td>
  </tr>
 </table>
</details>

<details>
  <summary>TODO</summary>
  
  <video src="enterurlhere" type="video/mp4">
</details>

### CODE REVIEW

**Comment by MrPeterss** [comment]
Make sure to give your pr a relevant title!

**Review by MrPeterss — CHANGES_REQUESTED** [review]
Nice work getting this out quick charles! However, there seems to be a bug where the "Meal Swipes" filter overrides the "BRB" one when they are both selected. You should decide with Thomas/designers if you want to override one of the filters when the other is selected. If this is the case, then you should visually show this by unselecting the filters. I had to do this for Compare Menus, and made a subclass of the EateryFilterViewController called CompareMenusFilterViewController. The other option is you can compound the two filters together (which would show eateries that accept brbs and meal swipes). Some of the other filters do this, so take a look at that!

File: Eatery Blue/UI/EateryFilter/EateryFilterViewController.swift [hunk 1]
```diff
@@ -55,14 +57,20 @@ class EateryFilterViewController: UIViewController {
...
+        filtersView.addButton(brbs)
+        setUpBRBs()
+        
         filtersView.addButton(favorites)
         setUpFavorites()
+        
+        
+        filtersView.addButton(under10Minutes)
+        setUpUnder10Minutes()
+
```
Comment (rs929) [inline Eatery Blue/UI/EateryFilter/EateryFilterViewController.swift:73]: Remove this Line

File: Eatery Blue/UI/EateryFilter/EateryFilterViewController.swift [hunk 2]
```diff
@@ -109,37 +117,48 @@ class EateryFilterViewController: UIViewController {
...
+            guard let self else { return }
 
-    private func setUpPaymentMethods() {
-        paymentMethods.label.text = "Payment Methods"
-        paymentMethods.imageView.isHidden = false
-        paymentMethods.tap { [weak self] _ in
+            allFiltersCallback?()
+            filter.brbsEnabled.toggle()
+            updateFilterButtonsFromState(animated: true)
+            delegate?.eateryFilterViewController(self, filterDidChange: filter)
```
Comment (rs929) [inline Eatery Blue/UI/EateryFilter/EateryFilterViewController.swift:144]: Add newline here

File: Eatery Blue/UI/EateryFilter/EateryFilterViewController.swift [hunk 3]
```diff
@@ -109,37 +117,48 @@ class EateryFilterViewController: UIViewController {
...
-            viewController.setUpSheetPresentation()
-            viewController.setSelectedPaymentMethods(filter.paymentMethods, animated: false)
-            viewController.delegate = self
-            if self.viewController != nil {
-                self.viewController?.present(viewController, animated: true)
-            } else {
-                tabBarController?.present(viewController, animated: true)
+            filter.under10MinutesEnabled.toggle()
+            updateFilterButtonsFromState(animated: true)
+            delegate?.eateryFilterViewController(self, filterDidChange: filter)
```
Comment (rs929) [inline Eatery Blue/UI/EateryFilter/EateryFilterViewController.swift:159]: add newline here

**Review by rs929 — APPROVED** [review]
Just a few small styling changes, Make sure to address Peter's changes!

Great PR!

**Comment by Xhether** [comment]
I fixed the filter bug

**Review by MrPeterss — APPROVED** [review]
Nice work! I'll show you how to resolve the merge conflict in work session! Wait to merge until then

**Pull request merged**